### ISSUE TITLE
Release 0.20.1 not SemVer compatible?

### ISSUE THREAD

**Issue by shaobo-he-aws** [lalrpop/lalrpop#848]
Hello,

Our project [Cedar](https://github.com/cedar-policy/cedar) depends on `lalrpop-util`. Updating to 0.20.1 causes our code to panic.  To reproduce,
```
git clone https://github.com/cedar-policy/cedar.git
cd cedar/cedar-policy-cli/sample-data/sandbox_a
cargo run evaluate \                               
    --principal 'User::"alice"' \
    --action 'Action::"listPhotos"' \
    --resource 'Album::"jane_vacation"' \
    --entities entities.json \
   "resource in Account::\"jane\""
```

The error message is,

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: BuildError { kind: NFA(BuildError { kind: Syntax(Translate(Error { kind: UnicodeNotAllowed, pattern: "(?:\"((?:(?:\\\\[\0-\t\u{b}-\u{10ffff}])|[\0-!\\#-\\[\\]-\u{10ffff}]))*\")", span: Span(Position(o: 19, l: 1, c: 20), Position(o: 23, l: 1, c: 21)) })) }) }', /private/tmp/cedar/target/debug/build/cedar-policy-core-894cc3fee656509a/out/src/parser/grammar.rs:57330:76
```

We need to enable `unicode` feature of `lalrpop-util` to avoid the crash, which seems to be a breaking change.

**Comment by dburgener** [comment]
Thanks for the report!  I guess we should yank the release, @youknowone?

Looking at the issue, I think I know what's going on here (this was initially very surprising to me, since I thought we got this exact issue sorted out before the release (#844)).

Checking your `cargo tree -e features`, I see that the `unicode` feature is being enabled in lalrpop, but not lalrpop-util.  That's happening because it's in the default features for lalrpop, but not lalrpop util.  We should definitely get consistent in setting those default features.  I think that had we done so, you would have picked up the default feature in util and this wouldn't have broken you.

I think that the state today is that lalrpop and lalrpop-util need to be consistent in terms of whether they use the unicode feature (I suspect 0.20.1 introduced this implicit requirement, but I'm not sure).  I'm not aware of if Cargo gives us a way to enforce that, but at least setting the defaults the same should avoid the pitfall.

I'll submit a PR to add unicode to the default features for lalrpop-util.  That would have caught this case.  I guess the release would still break anyone who was explicitly setting lalrpop vs lalrpop-util differently though.

**Comment by shaobo-he-aws** [comment]
Thank you for your quick response. We're looking forward to seeing the fixed version.

**Comment by youknowone** [comment]
@dburgener yanked 0.20.1

**Comment by oxalica** [comment]
> @dburgener yanked 0.20.1

Will it be republished as 0.21.0? 0.20.0 still has some `regex` feature issues unfixed.

**Comment by dburgener** [comment]
I think that once we merge #849, we'll be semver compatible with 0.20 and can republish as 0.20.2.  The semver issue here was more of a bug than an intentional change, so my preference is for rereleasing in a semver compatible fashion.

That said, we've had far too many issues around the whole regex/unicode thing, so I want to 1. Make sure we have maintainer consensus that we're not semver breaking after #849 and 2. Enhance our test coverage around testing proposed releases against user code bases.  I intend to drive both of those things to a close, but I have some other responsibilities at work this week eating up my time, so it will probably be another 2-3 weeks until we're ready to retry on this release.

**Comment by mmatsa** [comment]
It looks like #849 was merged on January 8.

Version 0.20.0 still depends on `is-terminal` which depends on an old version of `rustix` without the fix for the issue initially found in https://github.com/imsnif/bandwhich/issues/284

Therefore, version 0.20.0 is susceptible to:
> rustix's `rustix::fs::Dir` iterator with the `linux_raw` backend can cause memory explosion 

The latest code on `master` no longer has the dependency on `is-terminal` and is therefore not susceptible, and also with the fix for #849 on January 8 it also no longer has the problem opened by this issue #848.

Perhaps publishing a new 0.20.2 would have lalrpop no longer susceptible.  Is that planned?

**Comment by sunfishcode** [comment]
In case it simplifies anything, I can confirm that `is-terminal`'s use of `rustix` is not susceptible to the `rustix::fs::Dir` bug. `is-terminal` doesn't use `rustix::fs::Dir` at all.
